modules/compiler/riscv/source/riscv_pass_machinery.cpp

```diff
@@ -223,6 +223,11 @@ llvm::ModulePassManager RiscvPassMachinery::getLateTargetPasses() {
   // functions will be cloned to give a version using degenerate sub-groups and
   // a version using non-degenerate sub-groups, for selection by the runtime.
   tuner.degenerate_sub_groups = true;
+
+  // We run the handle barriers pass, which deals with work group reductions and
+  // broadcasts itself.
+  tuner.handling_barriers = true;
```

Comment:
Since this is only used in once place and the flag is `scansOnly` - should this flag be that - and a better description added? The default should be whatever works for the old main wg targets (host/riscv G1 WG, M1), so it's not a major change for any customers who would need to set it.

Reply:
i wanted the flag in the tuner to be as abstract as possible, the "scans only" meaning in the actual implementation is a temporary measure. Well, probably "handling barriers" should be too although perhaps there are other ways the vectorizer could use this information in future, who knows

Reply:
well it's open source so I'd rather it still reflected what it did with a big old comment.

Reply:
if true is default we don't need this line anymore anyway